### PR TITLE
Add process number to pool

### PR DESCRIPTION
**Pull request by JamesHuf**
Adds the 'Process Number' field to the pool model and the view and edit pool pages.

To test:
1. Visit `/en/admin/pools/{poolId}/edit` and confirm it looks ok.
2. Add a process number to the pool and ensure it updates properly.
3. Navigate to `/en/admin/pools/{poolId}` and confirm it displays the new process number for the pool.

Note: This PR adds untranslated strings to an existing page. How do we want to handle this?

### CODE REVIEW

File: api/graphql/schema.graphql [hunk 1]
```diff
@@ -1043,6 +1045,7 @@ input CreatePoolInput {
     operationalRequirements: [OperationalRequirement] @rename(attribute: "operational_requirements")
     keyTasks: LocalizedStringInput @rename(attribute: "key_tasks")
     status: PoolStatus = NOT_TAKING_APPLICATIONS @rename(attribute: "pool_status")
+    priorityNumber: String
```
Comment (brindasasi) [inline api/graphql/schema.graphql:1048]: priorityNumber? probably typo?

File: api/storage/app/lighthouse-schema.graphql [hunk 2]
```diff
@@ -317,6 +317,7 @@ input CreatePoolInput {
   operationalRequirements: [OperationalRequirement]
   keyTasks: LocalizedStringInput
   status: PoolStatus = NOT_TAKING_APPLICATIONS
+  priorityNumber: String
```
Comment (brindasasi) [inline api/storage/app/lighthouse-schema.graphql:320]: should it be processNumber instead?

**Review by brindasasi — CHANGES_REQUESTED** [review]

File: api/graphql/schema.graphql (hunk 1 above)
Comment (JamesHuf) [reply api/graphql/schema.graphql:1048]: Haha, yeah. They are both pr...number.

File: api/app/Models/Pool.php [hunk 3]
```diff
@@ -27,6 +27,7 @@
  * @property string $advertisement_language
  * @property boolean $is_published
  * @property string $stream
+ * @property string #process_number
```
Comment (vd1992) [inline api/app/Models/Pool.php:30]: does the "#" not break anything?

Comment (JamesHuf) [reply api/app/Models/Pool.php:30]: Oops. No, this is just documentation. It's not required at all.

**Comment by brindasasi** [comment]
After saving the process number and going back to pool again doesn't retrieve the process number. The value got saved in DB but the UI shows empty

File: api/graphql/schema.graphql (hunk 1 above)
Comment (mnigh) [reply api/graphql/schema.graphql]: In the issue linked to this PR, one of the [screenshots](https://user-images.githubusercontent.com/16977254/184662703-182b9ef8-0737-47d0-a718-c984055ccebb.png) shows a value with hyphens and alphabetical characters and the description of the issue describes it as "a string of characters", so `String` might actually be the correct type here. @Jerryescandon, can you confirm?

**Comment by JamesHuf** [comment]
> After saving the process number and going back to pool again doesn't retrieve the process number. The value got saved in DB but the UI shows empty

Interesting it's updating for me

**Comment by vd1992** [comment]
I think you should add this into the pool Cypress test, just for thorough coverage of that page
`frontend\cypress\integration\admin\pools.spec.js`

**Comment by vd1992** [comment]
> After saving the process number and going back to pool again doesn't retrieve the process number. The value got saved in DB but the UI shows empty

I can't seem to replicate this

**Comment by brindasasi** [comment]
> > After saving the process number and going back to pool again doesn't retrieve the process number. The value got saved in DB but the UI shows empty
> 
> I can't seem to replicate this



> > After saving the process number and going back to pool again doesn't retrieve the process number. The value got saved in DB but the UI shows empty
> 
> I can't seem to replicate this

This scenario works now. 
Below one doesn't work yet. 

1. Edit the pool 
2. Clear the process number
3. Save and reopen it
4. Expect to see the process number got cleared. But it doesn't. It is still there.

**Comment by JamesHuf** [comment]
> > > After saving the process number and going back to pool again doesn't retrieve the process number. The value got saved in DB but the UI shows empty
> > 
> > 
> > I can't seem to replicate this
> 
> This scenario works now. Below one doesn't work yet.
> 
> 1. Edit the pool
> 2. Clear the process number
> 3. Save and reopen it
> 4. Expect to see the process number got cleared. But it doesn't. It is still there.

This should be fixed now

**Comment by brindasasi** [comment]
> > > > After saving the process number and going back to pool again doesn't retrieve the process number. The value got saved in DB but the UI shows empty
> > > 
> > > 
> > > I can't seem to replicate this
> > 
> > 
> > This scenario works now. Below one doesn't work yet.
> > 
> > 1. Edit the pool
> > 2. Clear the process number
> > 3. Save and reopen it
> > 4. Expect to see the process number got cleared. But it doesn't. It is still there.
> 
> This should be fixed now

awesome work .. thank you!

**Review by brindasasi — APPROVED** [review]
LGTM

**Pull request merged**